Watchman.AwsResources/Services/AutoScaling/AutoScalingGroupAlarmDataProvider.cs

```diff
@@ -21,17 +33,62 @@ private static string GetAttribute(AutoScalingGroup resource, string property)
             }
         }
 
-        public decimal GetValue(AutoScalingGroup resource, string property)
+        public async Task<decimal> GetValue(AutoScalingGroup resource, AutoScalingResourceConfig config, string property)
         {
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+            
             switch (property)
             {
                 case "GroupDesiredCapacity":
-                    return resource.DesiredCapacity;
+                    return await GetGroupDesiredValue(resource, config);
             }
 
             throw new Exception("Unsupported property name");
         }
 
+        private async Task<decimal> GetGroupDesiredValue(AutoScalingGroup resource, AutoScalingResourceConfig config)
+        {
+            if (config.InstanceCountIncreaseDelayMinutes == null)
+            {
+                return resource.DesiredCapacity;
+            }
+
+            var delaySeconds = config.InstanceCountIncreaseDelayMinutes.Value * 60;
+            var now = _timeProvider.Now;
+
+            var metric = await _cloudWatch.GetMetricStatisticsAsync(
+                new GetMetricStatisticsRequest()
+                {
+                    Dimensions = new List<Dimension>()
+                    {
+                                new Dimension()
+                                {
+                                    Name = "AutoScalingGroupName",
+                                    Value = resource.AutoScalingGroupName
+                                }
+                    },
+                    Statistics = new List<string>() { "Minimum" },
+                    Namespace = "AWS/AutoScaling",
+                    Period = delaySeconds,
+                    MetricName = "GroupDesiredCapacity",
+                    StartTime = now.AddSeconds(-1 * delaySeconds),
+                    EndTime = now
```

Comment:
Do we prefer `DateTime.Now`, `DateTime.UtcNow` or `DateTimeoffset.UtcNow` ?

Reply:
I don't think we can use `DateTimeOffset` because the cloudwatch part of the aws sdk takes `DateTime`s.

But I guess `DateTime.UtcNow` would be better than `DateTime.Now` as being concerned with timezones is not important - both should represent the same point in time but `UtcNow` is simpler. Is that correct?

Reply:
I think that's correct.

Reply:
Actually that assumes that AWS are calling `.ToUniversalTime()` in the case that it's a local time, which may well not be the case. It would be better to be writing this code in the summer when we are BST and there is actually a difference in offsets...

Reply:
I will update to UtcNow

Reply:
Updated